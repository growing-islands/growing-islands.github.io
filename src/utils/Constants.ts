type BreakpointsType = {
  PC: number
  SP: number
}

export const BREAKPOINTS: BreakpointsType = {
  PC: 769,
  SP: 768,
}

export const PAGE_INFO = {
  URL: process.env.NEXT_PUBLIC_SITE_URL || 'https://',
  TWITTER: 'https://twitter.com/yumelab1214',
  FACEBOOK: 'https://www.facebook.com/yumelab.co.jp/',
}

export const INTERSECTION_OPTIONS = {
  root: null,
  rootMargin: '0px 0px -100%',
  threshold: 0,
}
