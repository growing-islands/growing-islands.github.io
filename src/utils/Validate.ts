/**
 * Form Validate関連の変数
 */

export const ERROR_MESSAGES = {
  required: 'この項目は入力必須です',
  maxLength: '入力できる文字数を超えています',
  mixLength: '文字数が足りません',
  max: '数値が大きすぎます',
  mix: '数値が小さすぎます',
  num: '半角数字で入力して下さい',
  hiragana: 'ひらがなで入力してください',
  fullKana: '全角カタカナで入力してください',
  tel: '電話番号が正しい形式ではありません',
  email: 'メールアドレスの形式が正しくありません',
  password:
    '英文字（大文字、小文字）、数字をそれぞれ最低一文字以上入れた8〜16文字で入力してください',
  dependent_char: '使用できない文字列が含まれています',
  url: '「http://」もしくは「https://」から始まるURLを入力してください',
}

// NOTE： 正規表現を記述するためにESLintを無効
/*eslint-disable no-useless-escape */
/*eslint-disable no-irregular-whitespace */
export const REGEX = {
  num: /^[0-9]+$/,
  alphanumeric: /^[0-9a-zA-Z]+$/,
  hiragana: /^[ぁ-んー　]*$/,
  fullKana: /^[ァ-ヶー　]*$/,
  halfKana: /^[ｦ-ﾟ]*$/,
  tel: /^(0{1}\d{9,10})$/,
  email:
    /^([\w!#$%&'*+\-\/=?^`{|}~]+(\.[\w!#$%&'*+\-\/=?^`{|}~]+)*|"([\w!#$%&'*+\-\/=?^`{|}~. ()<>\[\]:;@,]|\\[\\"])+")@(([a-zA-Z\d\-]+\.)+[a-zA-Z]+|\[(\d{1,3}(\.\d{1,3}){3}|IPv6:[\da-fA-F]{0,4}(:[\da-fA-F]{0,4}){1,5}(:\d{1,3}(\.\d{1,3}){3}|(:[\da-fA-F]{0,4}){0,2}))\])$/,
  password:
    /^(?=.*?[a-z])(?=.*?[A-Z])(?=.*?\d)[a-zA-Z0-9!"#$%&'()\*\+\-\.,\/:;<=>?@\[\\\]^_`{|}~]{8,16}$/,
  dependent_char:
    /^(?!.*(\u2460|\u2461|\u2462|\u2463|\u2464|\u2465|\u2466|\u2467|\u2468|\u2469|\u246a|\u246b|\u246c|\u246d|\u246f|\u2470|\u2471|\u2472|\u2473|\u2160|\u2161|\u2162|\u2163|\u2164|\u2165|\u2166|\u2167|\u2168|\u2169|\u3349|\u3314|\u3322|\u334d|\u3318|\u3327|\u3303|\u3336|\u3351|\u3357|\u330d|\u3326|\u3323|\u332b|\u334a|\u333b|\u339c|\u339d|\u339e|\u338e|\u338f|\u33c4|\u33a1|\u337b|\u301d|\u301f|\u2116|\u33cd|\u2121|\u32a4|\u32a5|\u32a6|\u32a7|\u32a8|\u3231|\u3232|\u3239|\u337e|\u337d|\u337c|\u222e|\u221f|\u22bf|\u2170|\u2171|\u2172|\u2173|\u2174|\u2175|\u2176|\u2177|\u2178|\u2179|\uffe2|\uffe4|[\uD800-\uDBFF][\uDC00-\uDFFF])).*$/,
  url: /^(https?)(:\/\/[-_.!~*\'()a-zA-Z0-9;\/?:\@&=+\$,%#]+)$/,
}
/*eslint-enable no-useless-escape */
/*eslint-enable no-irregular-whitespace */
