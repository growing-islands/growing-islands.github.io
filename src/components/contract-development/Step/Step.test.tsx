import { render } from '~tests/libs/utils'
import { Step } from '.'

describe('components/contract-development/Step', () => {
  it('Snapshot', () => {
    const { asFragment } = render(<Step />)

    expect(asFragment()).toMatchSnapshot()
  })
})
