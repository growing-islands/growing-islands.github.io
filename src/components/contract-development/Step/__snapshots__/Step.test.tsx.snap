// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/contract-development/Step Snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
    id="step"
  >
    <div
      class="l-wrap"
    >
      <div
        class="l-inner"
      >
        <div
          class="hidden tablet:block"
          data-testid="component"
        >
          <div
            class=""
          >
            <div>
              <span
                translate="no"
              >
                Step
              </span>
              <hr />
              <span>
                開発開始までの流れ
              </span>
            </div>
          </div>
        </div>
        <header
          data-aos="fade-up"
        >
          <h2
            class=""
            data-testid="component"
          >
            開発開始までの流れ
          </h2>
        </header>
        <ol
          data-aos="fade-up"
        >
          <li
            class=""
          >
            <div>
              <div>
                <img
                  alt="完成形の想像のイラスト"
                  placeholder="blur"
                  src="test-file-stub"
                />
              </div>
            </div>
            <div>
              <h3>
                完成形の想像
              </h3>
              <p>
                お客様側にて「どんな事を解決したいか」理想像を簡単におまとめ下さい。
              </p>
            </div>
            <svg
              aria-label="角"
              viewBox="0 0 21 31"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M.405.98h4.427l15.342 14.815L4.832 30.611H.527l15.347-14.816L.405.98Z"
              />
            </svg>
          </li>
          <li
            class=""
          >
            <div>
              <div>
                <img
                  alt="お問い合わせのイラスト"
                  placeholder="blur"
                  src="test-file-stub"
                />
              </div>
            </div>
            <div>
              <h3>
                お問い合わせ
              </h3>
              <p>
                フォームから、またはご紹介にて受付けを行なっております。
              </p>
            </div>
            <svg
              aria-label="角"
              viewBox="0 0 21 31"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M.405.98h4.427l15.342 14.815L4.832 30.611H.527l15.347-14.816L.405.98Z"
              />
            </svg>
          </li>
          <li
            class=""
          >
            <div>
              <div>
                <img
                  alt="初回のお打ち合わせのイラスト"
                  placeholder="blur"
                  src="test-file-stub"
                />
              </div>
            </div>
            <div>
              <h3>
                初回のお打ち合わせ
              </h3>
              <p>
                お客さまが検討されている事業の概要や現在の課題、将来のビジョンについて説明して頂きます。
              </p>
            </div>
            <svg
              aria-label="角"
              viewBox="0 0 21 31"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M.405.98h4.427l15.342 14.815L4.832 30.611H.527l15.347-14.816L.405.98Z"
              />
            </svg>
          </li>
          <li
            class=""
          >
            <div>
              <div>
                <img
                  alt="ご相談期間のイラスト"
                  placeholder="blur"
                  src="test-file-stub"
                />
              </div>
            </div>
            <div>
              <h3>
                ご相談期間
              </h3>
              <p>
                無料のご相談期間となります。
                <br
                  class="tablet:hidden"
                />
                (月4回まで)
                <br
                  class="hidden tablet:block"
                />
                事業の方向性を決定、実施すべき施策を可視化いたします。
              </p>
              <p
                class=""
                data-testid="component"
              >
                ※延長される場合は別途料金が発生する場合がございます
              </p>
            </div>
            <svg
              aria-label="角"
              viewBox="0 0 21 31"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M.405.98h4.427l15.342 14.815L4.832 30.611H.527l15.347-14.816L.405.98Z"
              />
            </svg>
          </li>
          <li
            class=""
          >
            <div>
              <div>
                <img
                  alt="正式契約のイラスト"
                  placeholder="blur"
                  src="test-file-stub"
                />
              </div>
            </div>
            <div>
              <h3>
                正式契約
              </h3>
              <p>
                まず3ヶ月以内で実現できる最低限のサービスを目指します。
                <br />
                毎週定例会を実施し、1週間単位で成果物を定義、実現していきます。
              </p>
            </div>
          </li>
        </ol>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
